people-and-planet-ai/land-cover-classification/predict_batch.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This does model predictions for different locations at a range of years."""
+
+import csv
+from typing import Iterable, List, NamedTuple, Optional, Sequence, Tuple
+
+import apache_beam as beam
+from apache_beam.io.filesystems import FileSystems
+from apache_beam.ml.inference.base import KeyedModelHandler
+from apache_beam.ml.inference.base import ModelHandler
+from apache_beam.ml.inference.base import RunInference
+from apache_beam.options.pipeline_options import PipelineOptions
+import numpy as np
+from typing_extensions import Self
```

Comment:
Seems like instead of using this, you could use 

```
from __future__ import annotations
```

and omit having to type hint `self`. Introduced in py3.7, but since this sample uses ==3.9 it should be good to use this. If you do so, you'd have to go down and make further changes to omit `self: Self` type hints.

I'd prefer using this to omit `self: Self` annotation - seems really really weird that we'd have to type hint these o__o;;

Reply:
After trying that, it looks like the linter still complains about the missing annotations.

Looking at the `flake8` documentation, it looks like `ANN101` is specifically meant for missing `self` annotations. They're not currently ignored in the linter, I think it's safe to ignore those as part of the default configuration, don't you think?

https://pypi.org/project/flake8-annotations/

I'll add `# noqa: ANN101` comments for now.